### PR TITLE
Parallel sections

### PR DESCRIPTION
**Pull request by lthakur007**
A test case which demonstrate that section inside "omp parallel sections" execute concurrently.

### CODE REVIEW

**Comment by lthakur007** [comment]
Hi @spophale ,
Please review this PR...

File: tests/4.5/parallel_sections/test_parallel_sections.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  while(1) {
+    pthread_mutex_lock(&Mutex);
+    if (*Var == 1) {
+      *Var += 1;
+      pthread_mutex_unlock(&Mutex);
+      break;
+    }
+  pthread_mutex_unlock(&Mutex);
+  }
+  pthread_mutex_unlock(&Mutex);
```
Comment (spophale) [inline tests/4.5/parallel_sections/test_parallel_sections.c:32]: redundant.

File: tests/4.5/parallel_sections/test_parallel_sections.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+// The following function checks if Var == 2 and then increments to Var = 3
+void function3(int *Var) {
+  while(1) {
+    pthread_mutex_lock(&Mutex);
+    if (*Var == 2) {
+      *Var += 1;
+      pthread_mutex_unlock(&Mutex);
```
Comment (spophale) [inline tests/4.5/parallel_sections/test_parallel_sections.c:68]: Why use pthread mutex and not omp atomic with seq_cst ?
Using the later will guarantee that the updated values are visible across all the threads of the parallel team.

File: tests/5.1/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_default.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  int Arr[32];
+  for (int i = 0; i < 32; ++i) {
+    Arr[i] = i;
+  }
+
+#pragma omp target teams distribute parallel for num_teams(2) thread_limit(10)\
+        default(private) map(to: Arr[0:32]) map(tofrom: ErrCount)
+  for (int i = 0; i < 32; ++i) {
+    CONST = 10;
+    Arr[i] += CONST;
```
Comment (spophale) [inline tests/5.1/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_default.c:47]: Same issues as pointed out in #783

**Review by spophale — APPROVED** [review]

**Review by fel-cab — APPROVED** [review]

**Pull request merged**